vignettes/tidyselect.Rmd

```diff
@@ -0,0 +1,310 @@
+---
+title: "Implementing tidyselect interfaces"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Implementing tidyselect interfaces}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(tidyselect)
+library(magrittr)
+```
+
+```{r, include = FALSE}
+# For better printing
+mtcars <- tibble::as_tibble(mtcars)
+iris <- tibble::as_tibble(iris)
+
+options(
+  tibble.print_min = 4,
+  tibble.print_max = 4
+)
+```
+
+tidyselect implements a specialised sublanguage of R for selecting
+variables from data frames and other data structures. A technical
+description of the DSL is available in the [syntax
+vignette](https://tidyselect.r-lib.org/articles/syntax.html). In this
+vignette, we describe how to include tidyselect in your own packages.
```

Comment:
If you just want to know how to use tidyselect syntax in dplyr or tidyr, please read xxxxxxxx instead.

Reply:
I'll add a TODO because we don't currently have a user-friendly description of the tidyselect syntax. Maybe we should make a cheatsheet instead of a vignette? Or both?

If we make a vignette, should it live in tidyselect or dplyr?

Reply:
hmm or we could link to this in the meantime?
https://r4ds.had.co.nz/transform.html#select